src/MDCT.jl

```diff
@@ -63,4 +67,60 @@ end
 imdct(X::AbstractVector{T}) where {T<:Number} =
     imdct(copy!(fftwsimilar(X, size(X)), X))
 
+mutable struct MDCTPlan{T<:fftwNumber, N, inv} <: AbstractFFTs.Plan{T}
+    plan::FFTW.r2rFFTWPlan{T} # plan_r2r! REDFT11 plan
+    pinv::AbstractFFTs.Plan{T}
+    MDCTPlan{T, N, inv}(plan) where {T, N, inv} = new(plan)
+end
+
+size(p::MDCTPlan{T, N, true}) where {T<:Number, N} = (div(N,2), N)
+size(p::MDCTPlan{T, N, false}) where {T<:Number, N} = (2*N, N)
+
+function mul!(Y::StridedArray{T}, p::MDCTPlan{T, N, false}, X::AbstractArray{T}) where {T<:Number, N}
+    @boundscheck (length(X) == 2*N && length(Y) == N) || throw(DimensionMismatch())
+    N2 = div(N,2)
+    @inbounds for i = 1:N2
+        Y[i] = -0.5 * (X[(3*N2+1)-i] + X[3*N2+i])
+        Y[N2+i] = -0.5 * (X[(2*N2+1)-i] - X[i])
+    end
+    return p.plan * Y
+end
+
+function mul!(Z::StridedArray{T}, p::MDCTPlan{T, N, true}, X::AbstractArray{T}) where {T<:Number, N}
+    @boundscheck length(Z) == N || throw(DimensionMismatch())
+    Y = p.plan*copy(X)
+    N1 = div(N,2)
+    N2 = div(N1,2)
+    s = 0.5 / N1
+    @inbounds for i = 1:N2
+        Z[N1+1-i] = -(Z[i] = Y[N2+i]*s)
+        Z[N1+N2+1-i] = (Z[N1+N2+i] = -Y[i]*s)
+    end
+    return Z
+end
+
+function *(p::MDCTPlan{T, N}, X::AbstractArray{T}) where {T<:Number, N}
+    Y = fftwsimilar(X, N)
+    return mul!(Y, p, X)
+end
+
+function plan_mdct(X::AbstractVector{T}) where {T<:Number}
```

Comment:
Should take `flags=FFTW.ESTIMATE, timelimit=Inf` keyword arguments.

Reply:
Done